tests/Tests.hs

```diff
@@ -105,9 +102,10 @@ instance Arbitrary Month where
 instance Arbitrary DateTime where
     arbitrary = DateTime <$> arbitrary <*> arbitrary
 instance Arbitrary Date where
-    arbitrary = Date <$> choose dateRange
-                     <*> arbitrary
-                     <*> choose (1,28)
+    arbitrary = do
+        year <- choose dateRange
+        month <- arbitrary
+        Date year month <$> choose (1, daysInMonth year month)
```

Comment:
I think Nicolas just missed the dependency; it's indeed the right thing to do here